### PR TITLE
EPROD-417 added checked_div and checked_mul to U256

### PR DESCRIPTION
**Pull request by veeso**
# Add Support for non-legacy transactions

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [x] Every function is properly unit tested
- [x] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/did/src/integer.rs [hunk 1]
```diff
@@ -8,7 +8,9 @@ use ic_stable_structures::{BoundedStorable, Storable};
 use num::BigUint;
 use serde::Serialize;
 
-#[derive(Debug, Default, Clone, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Hash)]
+#[derive(
+    Debug, Default, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Hash,
```
Comment (ufoscout) [inline src/did/src/integer.rs:12]: We explicitly wanted to avoid `copy` here

File: src/did/src/block.rs [hunk 2]
```diff
@@ -268,6 +269,36 @@ pub fn calculate_block_size<'a>(
     U256::from(block_size + transactions_size + size_field_size)
 }
 
+/// Calculate base fee for next block. [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md) spec
+pub fn calculate_next_block_base_fee(
```
Comment (ufoscout) [inline src/did/src/block.rs:273]: add tests for:
- "normal" values
- edge cases
- overflows

**Review by ufoscout — CHANGES_REQUESTED** [review]

**Review by itsyaasir — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]
